torch/csrc/jit/codegen/cuda/ir_base_nodes.h

```diff
@@ -480,19 +486,14 @@ class TORCH_CUDA_CU_API Expr : public Statement {
       std::vector<Val*> outputs,
       std::vector<Statement*> attributes);
 
+  virtual newObjectFuncType* newObjectFunc() const = 0;
```

Comment:
Why does this "newObject" interface now return a function pointer? What's wrong with the previous version?

Reply:
Because of this `container->removeExpr(op);` in the `mutate` method. When an IR node is removed from a container, its vtable no longer exist, and `op-> newObject` will cause segfault.

Reply:
Can we just do `op->newObject()` before removing it?

Reply:
And we can not swap the order of `op->newObject` and `container->removeExpr` because this causes some consistency check failure (an IterDomain used in multiple expr)

Reply:
OK, makes sense.